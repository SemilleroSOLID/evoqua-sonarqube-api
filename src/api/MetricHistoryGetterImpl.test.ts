import { MetricHistoryGetterImpl } from ".";

const { SONAR_URL, SONAR_USERNAME, SONAR_PASSWORD } = process.env;
const api = new MetricHistoryGetterImpl(SONAR_URL!);

beforeAll(() => api.login(SONAR_USERNAME!, SONAR_PASSWORD!));

it('getMetrics', async () => {
  const projectId = 'SCAPE-app';
  const metrics = await api.getMetrics(projectId);
  expect(metrics).toMatchSnapshot();
});

it('getMetricHistory', async () => {
  const projectId = 'SCAPE-app';
  const metric = {
    key: 'code_smells',
    name: 'Code Smells',
  };
  const metricHistory = await api.getMetricHistory(projectId, metric);
  expect(metricHistory).toMatchSnapshot();
});
