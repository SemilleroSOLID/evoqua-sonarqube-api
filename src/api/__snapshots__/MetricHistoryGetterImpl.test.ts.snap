// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMetricHistory 1`] = `
Object {
  "metric": "Code Smells",
  "values": Array [
    14,
    2,
    0,
  ],
  "versions": Array [
    "2021-05-07T09:42:24-0500",
    "2021-05-09T00:30:02-0500",
    "2021-07-29T23:39:01-0500",
  ],
}
`;

exports[`getMetrics 1`] = `
Array [
  Object {
    "key": "new_technical_debt",
    "name": "Added Technical Debt",
  },
  Object {
    "key": "code_smells",
    "name": "Code Smells",
  },
  Object {
    "key": "effort_to_reach_maintainability_rating_a",
    "name": "Effort to Reach Maintainability Rating A",
  },
  Object {
    "key": "sqale_rating",
    "name": "Maintainability Rating",
  },
  Object {
    "key": "new_maintainability_rating",
    "name": "Maintainability Rating on New Code",
  },
  Object {
    "key": "new_code_smells",
    "name": "New Code Smells",
  },
  Object {
    "key": "sqale_index",
    "name": "Technical Debt",
  },
  Object {
    "key": "sqale_debt_ratio",
    "name": "Technical Debt Ratio",
  },
  Object {
    "key": "new_sqale_debt_ratio",
    "name": "Technical Debt Ratio on New Code",
  },
]
`;
